tests/sentry/utils/sdk_crashes/test_sdk_crash_detection_dotnet.py

```diff
@@ -0,0 +1,226 @@
+from sentry.utils.sdk_crashes.sdk_crash_detection import sdk_crash_detection
+from sentry.utils.sdk_crashes.sdk_crash_detection_config import (
+    SDKCrashDetectionConfig,
+    build_sdk_crash_detection_configs,
+)
+
+
+def decorators(func):
+    @wraps(func)
+    @django_db_all
+    @pytest.mark.snuba
+    @patch("random.random", return_value=0.1)
+    @patch("sentry.utils.sdk_crashes.sdk_crash_detection.sdk_crash_detection.sdk_crash_reporter")
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+@pytest.fixture
+def configs() -> Sequence[SDKCrashDetectionConfig]:
+    with override_options(
+        {
+            "issues.sdk_crash_detection.dotnet.project_id": 6,
+            "issues.sdk_crash_detection.dotnet.sample_rate": 0.6,
+            "issues.sdk_crash_detection.dotnet.organization_allowlist": [5],
+        }
+    ):
+        return build_sdk_crash_detection_configs()
+
+
+@pytest.mark.parametrize(
+    ["sdk_frame_module", "system_frame_module", "detected"],
+    [
+        # Standard .NET SDK frames
+        (
+            "Sentry.SentryClient",
+            "System.Threading.Tasks.Task",
+            True,
+        ),
+        (
+            "Sentry.AspNetCore.SentryMiddleware",
+            "Microsoft.AspNetCore.Http.HttpContext",
+            True,
+        ),
+        (
+            "Sentry.Extensions.Logging.SentryLogger",
+            "Microsoft.Extensions.Logging.ILogger",
+            True,
+        ),
+        # Unity SDK frames
+        (
+            "Sentry.Unity.SentryUnity",
+            "UnityEngine.MonoBehaviour",
+            True,
+        ),
+        # System frames only - should not be detected
+        (
+            "System.Threading.Tasks.Task",
+            "Microsoft.AspNetCore.Http.HttpContext",
+            False,
+        ),
+        # Non-Sentry SDK frames - should not be detected
+        (
+            "MyApp.SomeClass",
+            "System.Threading.Tasks.Task",
+            False,
+        ),
+        # Misspelled Sentry - should not be detected
+        (
+            "Sentri.SentryClient",
+            "System.Threading.Tasks.Task",
+            False,
+        ),
+    ],
+)
+@decorators
+def test_sdk_crash_is_reported_with_dotnet_paths(
+    mock_sdk_crash_reporter,
+    mock_random,
+    store_event,
+    configs,
+    sdk_frame_module: str,
+    system_frame_module: str,
+    detected: bool,
+):
+    event = store_event(
+        data=get_crash_event(
+            sdk_frame_module=sdk_frame_module, system_frame_module=system_frame_module
+        )
+    )
+
+    # Find the dotnet config (should be the last one added)
+    dotnet_config = None
+    for config in configs:
+        if config.sdk_name.value == "dotnet":
+            dotnet_config = config
+            break
+
+    assert dotnet_config is not None, "Dotnet config should be present"
+    dotnet_config.organization_allowlist = [event.project.organization_id]
```

Comment:
Thanks for actually looking for the correct config and not relying on an index 👍